File: tiq-e/others/2/main.py
from time import perf_counter


class Solution:
    def hammingDistance(self, x: int, y: int) -> int:
        m = x ^ y
        b = bin(m)
        c = b[2::]
        d = 0
        for i in str(c):
            if i == '1':
                d += 1
        return d


x = 2
y = 4

start = perf_counter()
result = Solution().hammingDistance(x, y)
end = perf_counter()
print("Duration: " + str(end - start))
print("Result: " + str(result))

File: tiq-e/others/2/main.rs
use std::time::Instant;

struct Solution;

impl Solution {
    pub fn hamming_distance(x: i32, y: i32) -> i32 {
        let mut z = x ^ y;
        let mut r = 0;
        while z != 0 {
            r += 1;
            z = z & (z - 1);
        }
        r
    }
}

fn main() {
    let x = 2;
    let y = 4;

    let start = Instant::now();
    let result = Solution::hamming_distance(x, y);
    let end = start.elapsed().as_micros();
    println!("Duration: {:?}", end);
    println!("Result: {:?}", result);
}
